File: ItalianRestaurantBackend/src/main/java/com/example/italianrestaurant/meal/MealDto.java
package com.example.italianrestaurant.meal;

import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class MealDto {
    @NotNull(message = "Name is mandatory")
    private String name;
    @NotNull(message = "Name is mandatory")
    private String name_pl;
    @NotNull(message = "Image Path is mandatory")
    private byte[] imgData;
    @NotNull(message = "Category is mandatory")
    private String category;
    private String description;
    private String description_pl;
    @NotNull(message = "Price is mandatory")
    private double price;
}

File: ItalianRestaurantFrontend/src/app/models/mealDto.ts
export class MealDto {
  name: string
  name_pl: string
  description: string
  description_pl: string
  price: number
  category: string
  image?: string


  constructor(name: string, name_pl: string, description: string, description_pl:string,  price: number, mealCategory: string, image?: string) {
    this.name = name;
    this.name_pl = name_pl;
    this.description = description;
    this.description_pl = description_pl;
    this.price = price;
    this.category = mealCategory;
    this.image = image;
  }
}
